src/azul/service/avro_pfb.py

```diff
@@ -188,6 +192,17 @@ def from_json(cls,
         id_ = _reversible_join('.', map(str, (name, id_, len(ids))))
         return cls(id=id_, name=name, object=object_)
 
+    @classmethod
+    def for_replica(cls, replica: MutableJSON, schema: JSON) -> Self:
+        name, object_ = replica['replica_type'], replica['contents']
+        cls._add_missing_fields(name, object_, schema)
+        # Note that it is possible for two distinct replicas to have the same
+        # entity ID. For example, replicas representing the DUOS registration
+        # of AnVIL datasets have the same ID as the replica for the dataset
+        # itself. Terra appears to combine PFB entities with the same ID
+        # into a single row.
+        return cls(id=replica['entity_id'], name=name, object=object_)
```

Comment:
I think there should be a FIXME comment referring to https://github.com/DataBiosphere/azul/issues/6139 here.